test/module/libs/common/result_test.cpp

```diff
@@ -158,6 +158,146 @@ TEST(ResultTest, ResultVoidError) {
                makeFailCase<Error<void>>(kErrorCaseMessage));
 }
 
+/**
+ * @given Result some value
+ * @when some_val function is invoked
+ * @then the value is returned
+ */
+TEST(ResultTest, SomeVal) {
+  Result<int, void> result = makeValue(5);
+  ASSERT_TRUE(result.some_val());
+}
+
+/**
+ * @given Result some error
+ * @when some_err function is invoked
+ * @then the error is returned
+ */
+TEST(ResultTest, SomeErr) {
+  Result<void, int> result = makeError(5);
+  ASSERT_TRUE(result.some_err());
+}
+
+/**
+ * @given Pair of results with some values
+ * @when and_res function is invoked
+ * @then result contains the last value
+ */
+TEST(ResultTest, AndResWithValVal) {
+  Result<int, void> result = makeValue(5);
+  Result<int, void> new_res = makeValue(4);
+  result.and_res(new_res).match([](Value<int> v) { ASSERT_EQ(4, v.value); },
+                                makeFailCase<Error<void>>(kErrorCaseMessage));
+}
+
+/**
+ * @given Pair of results: first with error, second with value
+ * @when and_res function is invoked
+ * @then result contains the first error
+ */
+TEST(ResultTest, AndResWithErrVal) {
```

Comment:
What about two errors case ?

Reply:
Considered them as negligible. Do you think it's important?

Reply:
Will it increase the test coverage?